src/core/pointcloud/qgspointcloudblockhandle.h

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************
+                         qgspointcloudblockhandle.h
+                         --------------------
+    begin                : March 2021
+    copyright            : (C) 2021 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOINTCLOUDBLOCKHANDLE_H
+#define QGSPOINTCLOUDBLOCKHANDLE_H
+
+#include <QObject>
+
+#include "qgspointcloudattribute.h"
+#include "qgstiledownloadmanager.h"
+
+#define SIP_NO_FILE
+
+class QgsPointCloudAttributeCollection;
+class QgsPointCloudBlock;
+
+/**
+ * \ingroup core
+ * \brief Base class for handling loading QgsPointCloudBlock asynchronously
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
+ *
+ * \since QGIS 3.20
+ */
+class CORE_EXPORT QgsPointCloudBlockHandle : public QObject
+{
+    Q_OBJECT
+  public:
+    //! Constructor
+    QgsPointCloudBlockHandle( const QString &dataType, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, QgsTileDownloadManagerReply *tileDownloadManagerReply );
+  signals:
+    //! Emitted when the block is loaded successfully
+    void blockLoadingSucceeded( QgsPointCloudBlock *block );
```

Comment:
Passing a pointer to a new instance that should be deleted by caller is not a great pattern with signals/slots: in theory there could be multiple receivers connected to the signal and then it's unclear who should delete the pointer. A better solution is to make the block owned by QgsPointCloudBlockHandle class as a member variable (ideally unique_ptr).